dcos/auth.py

```diff
@@ -0,0 +1,308 @@
+import jwt
+import getpass
+import time
+import sys
+from six.moves import urllib
+from six.moves.urllib.parse import urljoin, urlparse
+
+from dcos import config, http, util
+from dcos.errors import (DCOSAuthenticationException, DCOSException,
+                         DCOSHTTPException)
+
+
+def _get_auth_scheme(response):
+    """Return authentication scheme requested by server for
+       'acsjwt' (DC/OS acs auth) or 'oauthjwt' (DC/OS acs oauth) type
+
+    :param response: requests.response
+    :type response: requests.Response
+    :returns: auth_scheme
+    :rtype: str
+    """
+
+    if 'www-authenticate' in response.headers:
+        auths = response.headers['www-authenticate'].split(',')
+        scheme = next((auth_type.rstrip().lower() for auth_type in auths
+                       if auth_type.rstrip().lower().startswith("acsjwt") or
+                       auth_type.rstrip().lower().startswith("oauthjwt")),
+                      None)
+        if scheme:
+            scheme_info = scheme.split("=")
+            auth_scheme = scheme_info[0].split(" ")[0].lower()
+            return auth_scheme
+        else:
+            msg = ("Server responded with an HTTP 'www-authenticate' field of "
+                   "'{}', DC/OS only supports ['oauthjwt', 'acsjwt']".format(
+                       response.headers['www-authenticate']))
+            raise DCOSException(msg)
+    else:
+        msg = ("Invalid HTTP response: server returned an HTTP 401 response "
+               "with no 'www-authenticate' field")
+        raise DCOSException(msg)
+
+
+def _prompt_user_for_token(url, token_type):
+    """Get Token from user
+
+    :param url: url for user to go to
+    :type url: str
+    :param token_type: type of token to be received
+    :type token_type: str
+    :returns: token show to user by browser
+    :rtype: str
+    """
+
+    msg = "\n{}\n\n    {}\n\nEnter {} Token:".format(
+          "Please go to the following link in your browser:",
+          url,
+          token_type)
+    sys.stderr.write(msg)
+    sys.stderr.flush()
+    token = sys.stdin.readline().strip()
```

Comment:
I may not understand enough of the CLI architecture, but why don't we use Python's `input([prompt])` built-in?